### ISSUE TITLE
Fortran modules not working

### ISSUE THREAD

**Issue by dbroemmel** [mattgodbolt/compiler-explorer#1123]
Thanks for taking the Fortran language support live!

After a short test I noticed a problem on the live site that did not show up on my local instance:
If the source defines a *module* the compiler fails with
``
Fatal Error: Can't open module file 'math.mod0' for writing at (1): Permission denied
compilation terminated.
``
(I guess the *0* with *mod0* is not correct).

Reason for this: for any module defined, the Fortran compiler will create a *.mod* file. By default *.mod* files are create in the current working directory. This does not work on the live site (working directory of course being non-writable).

The solution would be to tell the compiler where to store/load modules from. For gfortran, this would be an ``-J <path>``. Ideally, this path should be the same temporary directory as for the output file. I haven't quite worked out if this can be achieved via an ``options`` key in the properties file with the path being a variable.

**Comment by dbroemmel** [comment]
While adding an ``options=-J/tmp`` key to ``fortran.default.properties`` removes the error (even as an option in the live site), it will cause problems with multiple users creating modules with identical names.

I assume the solution would be a compiler specific file ``lib/compilers/fortran.js`` that adds the ``-J`` option, possibly within a ``prepareArguments`` function? But I lack the expertise to even test that. I can't figure out when those compiler specific classes/objects would be found or called.

I tested adding a line
``
options = options.concat("-J",outputFilename.replace("/output.s",""));
``
to this function in ``lib/base-compiler.js`` and it does what I want. But this presumably is far from what would be a clean solution, even for a compiler specific function. And by no means should it go into the BaseCompiler since it will break all other compilers.

**Comment by partouf** [comment]
making a specific compiler script is not that hard, it might look intimidating from examples, but they all started out pretty simple. I think as an alternative to this -J, you could implement runCompiler() with something like

```
runCompiler(compiler, options, inputFilename, execOptions) {
     execOptions = this.getDefaultExecOptions();
     execOptions.customCwd = path.dirname(inputFilename);
     return super.runCompiler(compiler, options, inputFilename, execOptions);
}
```

I think...

**Comment by dbroemmel** [comment]
@partouf I think I've pieced it all together and it seems to work for me locally.

Using a custom working directory is the better alternative than using ``-J`` since it should work for all Fortran compilers, not just gfortran. Rather than intimidating, I found it a little confusing until I discovered your rather recent ``AddingALanguage.md`` explaining the process. The key element I was missing was the ``compilerType`` in the properties.

I guess I'll open another PR for the fix?

While I'm at changing the properties file for Fortran: can we add ifort at the same time? Is there a list of installed versions?
